vendor/golang.org/x/sys/unix/zsyscall_freebsd_arm64.go

```diff
@@ -414,6 +424,16 @@ func ptrace(request int, pid int, addr uintptr, data int) (err error) {
 
 // THIS FILE IS GENERATED BY THE COMMAND AT THE TOP; DO NOT EDIT
 
+func ptracePtr(request int, pid int, addr unsafe.Pointer, data int) (err error) {
+	_, _, e1 := Syscall6(SYS_PTRACE, uintptr(request), uintptr(pid), uintptr(addr), uintptr(data), 0, 0)
+	if e1 != 0 {
+		err = errnoErr(e1)
+	}
+	return
+}
+
+// THIS FILE IS GENERATED BY THE COMMAND AT THE TOP; DO NOT EDIT
+
 func Access(path string, mode uint32) (err error) {
 	var _p0 *byte
 	_p0, err = BytePtrFromString(path)
```

Comment:
The code patch appears to be adding two new functions that use unsafe pointers: `ioctlPtr` and `ptracePtr`. These new functions use the existing `Syscall` function under the hood, so their implementation seems sound. However, since they are using unsafe pointers, it's important to verify that their usage is safe and secure within the context of the program.

One suggestion for improvement would be to add comments explaining the usage and potential risks associated with these new functions, particularly for developers who may not have as much experience with unsafe pointers.